src/ConsulResponse.php

```diff
@@ -30,8 +34,13 @@ public function getStatusCode(): int
         return $this->status;
     }
 
-    public function json(): ?array
+    public function json()
     {
         return json_decode($this->body, true, 512, \JSON_THROW_ON_ERROR);
     }
+
+    public function isSuccessful(): bool
+    {
+        return \in_array($this->status, [self::HTTP_OK, self::HTTP_CREATED, self::HTTP_NO_CONTENT], true);
```

Comment:
I'm not sure theses constant are useful.

You can replace this with 
```suggestion
        return $this->status >= 200 && $this->status < 300;
```

Reply:
done